aptos-move/move-examples/redemption/tests/redemption_tests.move

```diff
@@ -0,0 +1,78 @@
+#[test_only]
+module redemption::redemption_tests {
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, Metadata};
+    use aptos_framework::object::{Self, Object};
+    use aptos_framework::primary_fungible_store;
+    use redemption::redemption;
+    use std::option;
+    use std::signer;
+    use std::string;
+    use aptos_framework::account;
+    use aptos_framework::aptos_account;
+    use aptos_framework::coin;
+    use aptos_framework::coin::Coin;
+    use aptos_framework::stake;
+
+    struct WrappedCoin {}
+
+    #[test(deployer = @0xcafe, operator = @0x123, user = @0x234)]
+    public entry fun test_e2e(deployer: &signer, operator: &signer, user: &signer) {
+        stake::initialize_for_test(&account::create_signer_for_test(@0x1));
+
+        let user_addr = signer::address_of(user);
+        let coins = create_coin_and_mint(deployer, 1000);
+        aptos_account::deposit_coins(user_addr, coins);
+        let (to, native_tokens) = create_fungible_asset_and_mint(deployer, 1000);
+        primary_fungible_store::deposit(signer::address_of(operator), native_tokens);
+        redemption::create_pool<WrappedCoin>(deployer, to);
+
+        let operator_addr = signer::address_of(operator);
+        assert!(primary_fungible_store::balance(operator_addr, to) == 1000, 0);
+        redemption::deposit_native<WrappedCoin>(operator, 1000);
+        assert!(primary_fungible_store::balance(operator_addr, to) == 0, 0);
+
+        assert!(coin::balance<WrappedCoin>(user_addr) == 1000, 0);
+        redemption::redeem<WrappedCoin>(user, 500);
+        assert!(coin::balance<WrappedCoin>(user_addr) == 500, 0);
```

Comment:
`assert!` statements for `user`'s before and after of their fungible asset balance are needed.